<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>RUDP_SR</class>
    <widget class="QMainWindow" name="RUDP_SR">
    <property name="geometry">
    <rect>
    <x>0</x>
    <y>0</y>
    <width>666</width>
    <height>842</height>
    </rect>
    </property>
    <property name="windowTitle">
    <string>MainWindow</string>
    </property>
    <widget class="QWidget" name="centralwidget">
    <widget class="QLabel" name="label">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>60</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Server IP Address</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_8">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>160</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Client IP Address</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_9">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>110</y>
        <width>211</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Server Port Number</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_10">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>260</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Global Timer</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_11">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>210</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Client Port Number</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_12">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>410</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Window Size</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_13">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>360</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Packet Size</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_14">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>310</y>
        <width>241</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Re-Transmission Count</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_15">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>510</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Requested File</string>
    </property>
    </widget>
    <widget class="QLabel" name="label_16">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>460</y>
        <width>191</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Buffer Length</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="server_ip">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>70</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>127.0.0.1</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="server_port">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>120</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>50125</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="client_ip">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>170</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>127.0.0.1</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="req_file">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>520</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>test.txt</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="buffer_len">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>470</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>20</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="window_size">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>420</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>10</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="packet_size">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>370</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>10024</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="retrans_count">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>320</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>3</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="global_timer">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>270</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>10</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="client_port">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>220</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>50126</string>
    </property>
    </widget>
    <widget class="QPushButton" name="server_button">
    <property name="geometry">
        <rect>
        <x>80</x>
        <y>630</y>
        <width>191</width>
        <height>51</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Start Server</string>
    </property>
    <property name="autoDefault">
        <bool>true</bool>
    </property>
    <property name="default">
        <bool>true</bool>
    </property>
    </widget>
    <widget class="QLabel" name="label_2">
    <property name="geometry">
        <rect>
        <x>130</x>
        <y>20</y>
        <width>431</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 16pt &quot;Myanmar Text&quot;;</string>
    </property>
    <property name="text">
        <string>Reliable UDP with Selective Repeat</string>
    </property>
    </widget>
    <widget class="QPushButton" name="client_button">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>630</y>
        <width>191</width>
        <height>51</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Start Client</string>
    </property>
    <property name="autoDefault">
        <bool>true</bool>
    </property>
    <property name="default">
        <bool>true</bool>
    </property>
    </widget>
    <widget class="QLabel" name="label_17">
    <property name="geometry">
        <rect>
        <x>100</x>
        <y>560</y>
        <width>241</width>
        <height>41</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 14pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string>Re-Transmission Time</string>
    </property>
    </widget>
    <widget class="QLineEdit" name="reTransTime">
    <property name="geometry">
        <rect>
        <x>390</x>
        <y>570</y>
        <width>181</width>
        <height>31</height>
        </rect>
    </property>
    <property name="text">
        <string>0.5</string>
    </property>
    </widget>
    <widget class="QLabel" name="result_label">
    <property name="geometry">
        <rect>
        <x>180</x>
        <y>700</y>
        <width>301</width>
        <height>81</height>
        </rect>
    </property>
    <property name="styleSheet">
        <string notr="true">font: 12pt &quot;MS Shell Dlg 2&quot;;</string>
    </property>
    <property name="text">
        <string/>
    </property>
    </widget>
    </widget>
    <widget class="QMenuBar" name="menubar">
    <property name="geometry">
    <rect>
        <x>0</x>
        <y>0</y>
        <width>666</width>
        <height>26</height>
    </rect>
    </property>
    </widget>
    <widget class="QStatusBar" name="statusbar"/>
    </widget>
    <tabstops>
    <tabstop>server_ip</tabstop>
    <tabstop>server_port</tabstop>
    <tabstop>client_ip</tabstop>
    <tabstop>client_port</tabstop>
    <tabstop>global_timer</tabstop>
    <tabstop>retrans_count</tabstop>
    <tabstop>packet_size</tabstop>
    <tabstop>window_size</tabstop>
    <tabstop>buffer_len</tabstop>
    <tabstop>req_file</tabstop>
    <tabstop>server_button</tabstop>
    </tabstops>
    <resources/>
    <connections/>
</ui>